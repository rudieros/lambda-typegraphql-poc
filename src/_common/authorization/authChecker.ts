export const authChecker = (input) => {
  // TODO implement auth logic here
  console.log('Auth CHecked input', input)
  return true
}
