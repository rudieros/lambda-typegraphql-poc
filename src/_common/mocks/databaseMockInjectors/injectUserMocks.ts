import { UserDB, UserEntity } from '../../database/entities/userEntity'

export const MOCK_USERS_ME: UserDB = {
  id: 'ned',
  email: 'ned@starks.com',
  name: 'Ned Stark',
}

export const MOCK_USERS = {
  me: MOCK_USERS_ME,
  arya: {
    id: 'arya',
    email: 'arya@starks.com',
    name: 'Arya Stark',
  } as UserDB,
  tywin: {
    id: 'tywin',
    email: 'tywin@lannister.com',
    name: 'Tywin Lannister',
  },
  jamie: {
    id: 'jamie',
    email: 'jamie@lannister.com',
    name: 'Jamie Lannister',
  },
  rooseBolton: {
    id: 'roose',
    email: 'roose@bolton.com',
    name: 'Roose Bolton',
  },
}

export const injectUserMocks = async () => {
  console.log('===> Will inject Users Mocks')
  await UserEntity.batchPut(
    Object.values(MOCK_USERS)
    // .map((user) => ({ ...user, sort: 'user' }))
  ).catch((e) => {
    console.log('E', e)
  })
}
