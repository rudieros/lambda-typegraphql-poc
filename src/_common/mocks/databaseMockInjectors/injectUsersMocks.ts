import { UserDB, UserEntity } from '../../database/entities/userEntity'

export const MOCK_USERS_ME: UserDB = {
  id: '1',
  sort: 'user',
  email: 'ned@starks.com',
  name: 'Ned Stark'
}

export const MOCK_USERS: UserDB[] = [
  MOCK_USERS_ME,
  {
    id: '2',
    sort: 'user',
    email: 'arya@starks.com',
    name: 'Arya Stark'
  }
]

export const injectUsersMocks = async () => {
  console.log('===> Will inject Users Mocks')
  await UserEntity.batchPut(MOCK_USERS)
}
