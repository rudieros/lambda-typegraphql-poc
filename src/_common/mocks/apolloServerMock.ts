import { createTestClient } from "apollo-server-testing";
import { ApolloServer } from "apollo-server-lambda";
import { schema } from "../../server";
import { UserRoles } from "../authorization/UserRoles";
import { Container } from "typedi";

export const createApolloTestClient = (config?: {
  uid?: string;
  userRole?: UserRoles;
}) =>
  createTestClient(new ApolloServer({
    schema,
    context: () => {
      const requestId = Math.floor(Math.random() * Number.MAX_SAFE_INTEGER); // uuid-like
      const container = Container.of(requestId); // get the scoped container
      const context = { requestId, container }; // create fresh context object
      container.set("context", context); // place context or other data in container
      return {
        ...context,
        uid: config && config.uid,
        userRole: config && config.userRole
      };
    }
  }) as any);
