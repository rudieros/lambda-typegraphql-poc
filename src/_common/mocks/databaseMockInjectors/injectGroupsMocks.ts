import { GroupDB, GroupEntity } from "../../database/entities/groupEntity";
import { MOCK_USERS } from "./injectUserMocks";

export const MOCK_GROUPS = {
  houseStark: {
    id: "nci39fhal",
    name: "House Stark",
    memberCount: 122,
    creatorId: MOCK_USERS.me.id,
    queryPartition1: MOCK_USERS.me.id,
    querySort1: "2019-07-29T02:21:24.982Z"
  } as GroupDB,
  lordsOfWinterfell: {
    id: "ij19dl10",
    name: "Lords of Winterfell",
    memberCount: 4,
    creatorId: MOCK_USERS.me.id,
    queryPartition1: MOCK_USERS.me.id,
    querySort1: "2019-07-29T02:20:42.069Z"
  } as GroupDB,
  houseLannister: {
    id: "k20dksl",
    name: "House Lannister",
    memberCount: 123,
    creatorId: MOCK_USERS.tywin.id,
    queryPartition1: MOCK_USERS.tywin.id,
    querySort1: "2019-07-29T02:20:42.069Z"
  } as GroupDB,
  manyFacePeople: {
    id: "i2kd91k",
    name: "Many Face People",
    memberCount: 1,
    creatorId: MOCK_USERS.arya.id,
    queryPartition1: MOCK_USERS.arya.id,
    querySort1: "2019-07-29T02:20:42.069Z"
  } as GroupDB
};

export const injectGroupMocks = async () => {
  console.log("===> Will inject Group Mocks");
  await GroupEntity.batchPut(Object.values(MOCK_GROUPS)).catch(e => {
    console.log("E", e);
  });
};
