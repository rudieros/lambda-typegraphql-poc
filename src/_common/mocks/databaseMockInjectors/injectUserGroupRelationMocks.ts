import { MOCK_USERS } from './injectUserMocks'
import {
  UserGroupRelationDB,
  UserGroupRelationsEntity,
} from '../../database/entities/userGroupRelationEntity'
import { MOCK_GROUPS } from './injectGroupsMocks'

export const MOCK_USER_GROUP_RELATIONS: UserGroupRelationDB[] = [
  {
    id: MOCK_USERS.me.id,
    sort: MOCK_GROUPS.houseStark.id,
    creationDate: '2019-07-29T02:21:24.982Z',
  },
  {
    id: MOCK_USERS.me.id,
    sort: MOCK_GROUPS.lordsOfWinterfell.id,
    creationDate: '2019-07-29T02:21:24.982Z',
  },
  {
    id: MOCK_USERS.arya.id,
    sort: MOCK_GROUPS.houseStark.id,
    creationDate: '2019-07-29T02:21:24.982Z',
  },
  {
    id: MOCK_USERS.arya.id,
    sort: MOCK_GROUPS.lordsOfWinterfell.id,
    creationDate: '2019-07-29T02:21:24.982Z',
  },
  {
    id: MOCK_USERS.arya.id,
    sort: MOCK_GROUPS.manyFacePeople.id,
    creationDate: '2019-07-29T02:21:24.982Z',
  },
  {
    id: MOCK_USERS.jamie.id,
    sort: MOCK_GROUPS.houseLannister.id,
    creationDate: '2019-07-29T02:21:24.982Z',
  },
  {
    id: MOCK_USERS.tywin.id,
    sort: MOCK_GROUPS.houseLannister.id,
    creationDate: '2019-07-29T02:21:24.982Z',
  },
]

export const injectUserGroupRelationMocks = async () => {
  console.log('===> Will inject UserGroupRelation Mocks')
  await UserGroupRelationsEntity.batchPut(MOCK_USER_GROUP_RELATIONS).catch(
    (e) => {
      console.log('E', e)
    }
  )
}
