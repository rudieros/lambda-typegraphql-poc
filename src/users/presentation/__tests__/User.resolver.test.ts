import gql from 'graphql-tag'
import { createApolloTestClient } from '../../../_common/mocks/apolloServerMock'
import { MOCK_USERS_ME } from '../../../_common/mocks/databaseMockInjectors/injectUserMocks'
import { UserRoles } from '../../../_common/authorization/UserRoles'

describe('UserResolver', () => {
  describe('Authed Queries', () => {
    const { query } = createApolloTestClient({
      uid: MOCK_USERS_ME.id,
      userRole: UserRoles.USER,
    })
    test('`me` should return logged user successfully', async () => {
      const ME = gql`
        query {
          me {
            id
            name
          }
        }
      `
      const result = await query({
        query: ME,
      })
      expect(result.data.me.id).toBe(MOCK_USERS_ME.id)
      expect(result.data.me.name).toBe(MOCK_USERS_ME.name)
      expect(result.data.me).toMatchSnapshot()
    })
  })
})
