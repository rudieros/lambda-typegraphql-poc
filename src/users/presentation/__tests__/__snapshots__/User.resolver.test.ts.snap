// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserResolver Authed Queries \`me\` should return logged user successfully 1`] = `
Object {
  "id": "ned",
  "name": "Ned Stark",
}
`;
